Library/Homebrew/formulary.rb

```diff
@@ -1004,7 +1004,7 @@ def self.tap_paths(name, taps = Tap)
   end
 
   def self.find_formula_in_tap(name, tap)
-    filename = "#{name}.rb"
+    filename = "#{name}#{".rb" unless name.end_with?(".rb")}"
```

Comment:
Fixes an issue where ".rb" would be unnecessarily added to a path argument that had it already.

Reply:
```suggestion
    filename = name
    filename << ".rb" unless filename.end_with?(".rb")
```

Reply:
Had to use `name.dup` otherwise tests like `brew audit --skip-style --except=version openssl@3` would fail.